src/CPK.ts

```diff
@@ -302,6 +316,63 @@ class CPK {
     })
   }
 
+  async getEnabledSafeModules(): Promise<Address[]> {
+    if (!this.#contract) {
+      throw new Error('CPK contract uninitialized')
+    }
+    const modules = await this.#contract.call('getModules', [])
+    return modules
+  }
+
+  async isSafeModuleEnabled(moduleAddress: Address): Promise<boolean> {
+    if (!this.#contract) {
+      throw new Error('CPK contract uninitialized')
+    }
+    const modules = await this.#contract.call('getModules', [])
+    const selectedModules = modules.filter(
+      (mod: Address) => mod.toLowerCase() === moduleAddress.toLocaleLowerCase()
+    )
+    return selectedModules.length > 0
+  }
+
+  async enableSafeModule(
+    moduleAddress: Address,
+    options?: ExecOptions
+  ): Promise<TransactionResult> {
+    if (!this.#contract) {
+      throw new Error('CPK contract uninitialized')
+    }
+    const ownerAccount = await this.getOwnerAccount()
+    if (!ownerAccount) {
+      throw new Error('CPK ownerAccount uninitialized')
+    }
+    const sendOptions = normalizeGasLimit({ ...options, from: ownerAccount })
+    const txResult = await this.#contract.send('enableModule', [moduleAddress], {
+      ...sendOptions,
+      from: ownerAccount
```

Comment:
I think `enableModule` follows the rule for `SelfAuthorized`:

```
require(msg.sender == address(this), "Method can only be called from this contract");
```

So essentially you would use `execTransaction` to call it.

Reply:
Yes, I'm updating this and pushing the tests :)